pkg/types/userconfig/api.go

```diff
@@ -201,6 +203,9 @@ func AutoscalingFromAnnotations(deployment kmeta.Object) (*Autoscaling, error) {
 	if err != nil {
 		return nil, err
 	}
+	if maxReplicaConcurrency > math.MaxUint16 {
```

Comment:
Nice! I hadn't realized that `math.MaxUint16 == 65535` :)

There is slightly better place to add this check, which I definitely would not have expected you to find: https://github.com/cortexlabs/cortex/blob/master/pkg/types/spec/validations.go#L277

You can add a something like:

```go
LessThanOrEqualTo: pointer.Int32(math.MaxUint16),
```